### PR TITLE
Add chirp interop flag

### PR DESCRIPTION
**Pull request by NathanFlurry**
Fixes CLI-47

### CODE REVIEW

**Comment by NathanFlurry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/299?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #298** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/298?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #299** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/299?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/299?utm_source=stack-comment).

**Review by AngelOnFira — APPROVED** [review]

File: proto/chirp.proto [hunk 1]
```diff
@@ -86,15 +87,6 @@ message Response {
...
 
-enum ErrorCode {
-	reserved 1002;
-	UNKNOWN = 0;
-
-	INTERNAL = 1001;
-	RECURSIVE_REQUEST = 1003;
-	TIMED_OUT = 1004;
-}
-
```
Comment (MasterPtato) [inline proto/chirp.proto:97]: Does this affect any of our grafana dashboards?

File: lib/chirp/worker/src/manager.rs [hunk 2]
```diff
@@ -935,12 +933,12 @@ where
 
 					let err_proto = Into::<chirp::response::Err>::into(err);
 
-					if let Some(chirp::response::err::Kind::Internal(error)) = &err_proto.kind {
+					if let Some(chirp::response::err::Kind::BadRequest(error)) = &err_proto.kind {
```
Comment (MasterPtato) [inline lib/chirp/worker/src/manager.rs:936]: When would a `BadRequest` error be recursive?

File: lib/chirp/worker/src/manager.rs [hunk 3]
```diff
@@ -758,7 +758,7 @@ where
 				if let Some(error) = &summary.error {
 					match &error.kind {
 						Some(chirp::response::err::Kind::Internal(error)) => {
-							let error_code_str = error.code.to_string();
+							let error_code_str = "__UNKNOWN__".to_string();
```
Comment (MasterPtato) [inline lib/chirp/worker/src/manager.rs:761]: Does this make all internal requests have an unknown error code in grafana?

**Review by MasterPtato — CHANGES_REQUESTED** [review]

**Comment by MasterPtato** [comment]
How were each of the threads resolved?

**Comment by NathanFlurry** [comment]
> How were each of the threads resolved?

I resolved them instinctively. I should've left them open mb.

**Comment by NathanFlurry** [comment]
@MasterPtato is this pr good to merge? needs an approval.

**Review by MasterPtato — APPROVED** [review]

**Pull request merged**